lib/_parse-gerber.js

```diff
@@ -28,8 +28,8 @@ var reUNITS = /^%MO(IN|MM)/
 var reFORMAT = /^%FS([LT]?)([AI]?)(.*)X([0-7])([0-7])Y\4\5/
 var rePOLARITY = /^%LP([CD])/
 var reSTEP_REP = /^%SR(?:X(\d+)Y(\d+)I([\d.]+)J([\d.]+))?/
-var reTOOL_DEF = /^%ADD0*(\d{2,})([A-Za-z_]\w*)(?:,((?:X?[\d.]+)*))?/
-var reMACRO = /^%AM([A-Za-z_]\w*)\*?(.*)/
+var reTOOL_DEF = /^%ADD0*(\d{2,})([A-Za-z_][\w\-\.]*)(?:,((?:X?[\d.]+)*))?/
+var reMACRO = /^%AM([A-Za-z_][\w\-\.]*)\*?(.*)/
```

Comment:
As mentioned in your PR summary, the first character may be a `$` and this regex should reflect that